optaplanner-wb-screens/optaplanner-wb-domain-editor/optaplanner-wb-domain-editor-backend/src/main/java/org/optaplanner/workbench/screens/domaineditor/backend/server/PlannerDomainHandler.java

```diff
@@ -66,15 +66,17 @@ public void setDefaultValues( DataObject dataObject, Map<String, Object> options
 
     @Override
     public void processDataObject( DataObject dataObject, DataModel dataModel ) {
-        boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
-        boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
-        if ( hasPlanningEntity && hasComparatorDefinition ) {
-            try {
-                ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
-                dataObject.getNestedClasses().clear();
-                dataObject.getNestedClasses().add( comparatorObject );
-            } catch ( Exception e ) {
-                logger.error( "Unable to extract comparator object from " + dataObject.getClassName() );
+        if ( dataModel != null ) {
+            boolean hasPlanningEntity = dataObject.getAnnotation( PlannerDomainAnnotations.PLANNING_ENTITY_ANNOTATION ) != null;
+            boolean hasComparatorDefinition = dataObject.getAnnotation( ComparatorDefinition.class.getName() ) != null;
+            if ( hasPlanningEntity && hasComparatorDefinition ) {
+                try {
+                    ComparatorObject comparatorObject = plannerDataObjectEditorService.extractComparatorObject( dataObject, dataModel );
+                    dataObject.getNestedClasses().clear();
+                    dataObject.getNestedClasses().add( comparatorObject );
+                } catch ( Exception e ) {
```

Comment:
Don't catch Exception or Throwable, that's overkill.
If any of that code throws a checked exception, catch that, optionally with RuntimeException too if needed.
If you want to catch 2 exceptions, use java 7 style "catch (FooException | BarException e)"

Catching Exception is bad because if the code has no checked exceptions, then you could just catch RuntimeException. If it has, you can catch those and optionally RuntimeException (if the latter makes sense and often it does not make sense to catch RuntimeException)
Catching Throwable would be even worse, as that catches OutOfMemoryError etc, which need to bubble up the stack.